### DISCUSSION TITLE
unexpected result when backprojecting a simple single projection with detector offset by non-integer number of pixel.

### DISCUSSION THREAD

**Discussion by lesaintjerome** [astra-toolbox/astra-toolbox#520] [Q&A]
Hi all,
I am new to Astra and am trying to implement a reconstruction algorithm that needs to account for vertical detector displacements and I encounter some unexpected results. My simple toy case is as follows:
```
toy = np.zeros((4,1,3))
toy[1:2,0,1:2] = 1

sh = 1.1
vec = np.array([[0,-1,0,0,0,sh,1,0,0,0,0,1]],dtype=np.float64)
proj_geom = astra.create_proj_geom('parallel3d_vec',4,3,vec)

vol_geom = vol_geom = astra.creators.create_vol_geom(3, 3,4)
id, volume = astra.create_backprojection3d_gpu(toy.copy(), proj_geom, vol_geom)
```
If I set `sh=1.` (or any integer number), result is fine:; the non-zero pixel is shifted vertically in the reconstructed volume by the desired (integer) number of pixels and its value is 1.
If I set `sh=1.1`, I would expect in the backprojected volume a value of `0.9` in the voxel (2,1) and `0.1` in the voxel (3,1), due to the necessary interpolation.
Instead, I get `0.898` and `0.102` respectively. Which is a few percents discrepancy. It happens that this discrepancy seems to be amplified when I do a full MLEM reconstruction (with Gibbs-like oscillations at the edges of my object).

Is this discrepancy in the backprojection an exepected behavior? Did I do somtething wrong? 
I searched the discussions for something similar, in vain.
Thanks in advance for your help,

Jérome

**Comment by askorikov** [comment]
Hi! This is most likely because of the low precision (8-bit) of texture interpolation in CUDA, which gets amplified during iterations of MLEM. If you really need to fix this, you can shift the projection data instead of shifting Astra geometry, which should be equivalent for parallel geometries.

**Reply by lesaintjerome** [reply]
Hi thanks for the reply. It is indeed what we did.
